enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,451 @@
+* Provide manifests for quickly deploying the operator (including a minimal ExternalDNS Operator CR manifest).
+* Create tooling to simplify ExternalDNS troubleshooting. For example, add support for the Operator to
+OpenShift [must-gather](https://github.com/openshift/must-gather).
+* Integrate ExternalDNS with the OpenShift monitoring toolchain. For example, add
+[metrics support](https://github.com/openshift/cluster-monitoring-operator) for the operator and its operand(s).
+
+
+In addition, the following functionality is expected as part of the operator (note that this is not an exhaustive list):
+
+* Introduce a CRD to be used by the operator to manage ExternalDNS and its dependencies.
+* Common fields in spec that define the configuration of ExternalDNS and any dependent resources,
+i.e. a secret resource containing credentials for the necessary cloud provider.
+* Common fields in spec that define how to deploy ExternalDNS.
+* Common fields in status that expose the current health & version information of ExternalDNS operands.
+* The operator will be capable of observing other resources to perform basic sequencing required for OpenShift
+integration.
+* The operator will be capable of deploying multiple instances of ExternalDNS (operands), which is necessary when
+a common domain is shared across DNS zones, or when using multiple DNS Providers. See ExternalDNS' upstream
+[AWS public/private zone tutorial docs](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/public-private-route53.md)
+as an example.
+
+Here is what a CRD instance for ExternalDNS Operator may look like:
+
+```yaml
+apiVersion: operator.externaldns.io/v1alpha1
+kind: externaldns
+metadata:
+  name: default
+  namespace: my-externaldns-namespace
+spec:
+  providerCredentials: my-secret-reference
+  DNSZones:
+    - my-public-zone
+    - my-private-zone
+  domains:
+    - my-cluster.my-domain.com
+  provider: AWS
+  sources:
+    - route
+    - CRD
+status:
+  conditions:
+  - lastTransitionTime: "2020-08-20T23:01:33Z"
+    reason: Valid
+    status: "True"
+    type: Admitted
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    status: "True"
+    type: Available
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    message: The deployment has Available status condition set to True
+    reason: DeploymentAvailable
+    status: "False"
+    type: DeploymentDegraded
+```
+
+Additional `spec` and/or `status` fields may be introduced based on requirements.
+
+By exposing broad spec fields such as `DNSZones` and `provider`, ExternalDNS operator users would have the ability to
+enable ExternalDNS for multiple providers across multiple DNS zones. In general, a single instance of the ExternalDNS Operator
+CRD would correlate to a single [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) of ExternalDNS.
+
+On OpenShift, the ExternalDNS Operator could create a `credentialsrequest.cloudcredential.openshift.io` resource so that
```

Comment:
can you clarify what the permissions are required across the set of targeted providers?

for aws in particular, is there any issue aligning with sts-only clusters
https://docs.openshift.com/container-platform/4.7/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#sts-mode-installing-manual-run-installer

Reply:
Clarified permissions in the enhancement.

I'll add a note about adding STS support. Thanks!